File: backend/src/main/java/net/tape/model/TickerQuote.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class TickerQuote implements Serializable {
    private Long id;
    private String symbol;
    private String price;
    private String change;
    private String direction;
    private int sort;
}

File: web/src/api/components/schemas/TickerQuote.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type TickerQuote = {
	id: Option<Int>;
	symbol: Option<string>;
	price: Option<string>;
	change: Option<string>;
	direction: Option<string>;
	sort: Option<Int>;
};
export const TickerQuoteIO = type(
	{
		id: optionFromNullable(Int),
		symbol: optionFromNullable(string),
		price: optionFromNullable(string),
		change: optionFromNullable(string),
		direction: optionFromNullable(string),
		sort: optionFromNullable(Int),
	},
	'TickerQuote',
);
